share/spice/libraries/libraries.js

```diff
@@ -0,0 +1,207 @@
+(function(env) {
+    "use strict";
+    var search_items = DDG.get_query().split(" ");
+
+    var supported_package_managers = {
+        alcatraz: "http://alcatraz.io/",
+        atom: "https://atom.io/packages",
+        bower: "https://bower.io/",
+        cran: "https://cran.r-project.org",
+        dub: "https://code.dlang.org/download",
+        elm: "http://elm-lang.org/",
+        emacs: "https://www.gnu.org/software/emacs/",
+        go: "https://golang.org/",
+        inqlude: "https://inqlude.org/",
+        julia: "http://pkg.julialang.org/",
+        meteor: "https://www.meteor.com/",
+        nimble: "https://nim-lang.org/",
+        nuget: "https://www.nuget.org/",
+        shards: "https://crystal-shards-registry.herokuapp.com/",
+        wordpress: "https://wordpress.org/",
+    }
+
+    var install_scripts = {
+        bower: "$ bower install",
+        cran: "install.packages('",
+        elm: "$ elm-package install",
+        go: "$ go get",
+        julia: "Pkg.add(\"",
+        meteor: "meteor npm install --save",
+        nimble: "$ nimble install",
+        nuget: "PM> Install-Package",
+        pub: "dependencies:\n\t\t",
+        shards: "common_mark:\n\t\tgithub: ysbaddaden/crystal-cmark"
+    }
+
+    function getInstallScript(pkg_manager, pkg_name, pkg_version) {
+        var index = pkg_manager.toLowerCase();
+
+        if(index === "cran") {
+            return install_scripts[index] + pkg_name + "')";
+        } else if(index === "julia") {
+            return install_scripts[index] + pkg_name + "\")";            
+        } else if(index === "nuget") {
+            return install_scripts[index] + " " + pkg_name + " -Version " + pkg_version;
+        } else if(index === "pub") {
+            return install_scripts[index] + pkg_name + ": ^" + pkg_version;
+        } else if(install_scripts[index] === undefined) {
+            return false;
+        } else {
+            return install_scripts[index] + " " + pkg_name;
+        }
+    }
+
+    env.ddg_spice_libraries = function(api_result) {
+        
+        if(!api_result || api_result.length === 0) {
+            return Spice.failed('libraries');
+        }
+
+        var tile_template =  {
+            group: 'text',
+            detail: false,
+            item_detail: false,
+            variants: {
+                tile: 'basic1',
+                tileTitle: '1line',
+                tileFooter: '2line',
+                tileSnippet: 'large'
+            },
+            options: {
+                footer: Spice.libraries.footer,
+                moreAt: true
+            }
+        }
+        
+        var text_template =  {
+            group: 'text',
+            options: {
+                content: Spice.libraries.content,
+                moreAt: true
+            }
+        }
+
+        var template, meta, signal;
```

Comment:
I would initialize template and signal to `tile_template` and `low`. Then you only need to set it for special cases.